plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -276,10 +276,10 @@ public void testReadWholePartition()
     @Test
     public void testSelfJoin()
     {
-        assertUpdate("CREATE TABLE test_self_join_table AS SELECT 2 as age, 0 parent, 3 AS id", 1);
+        assertUpdate("CREATE TABLE test_self_join_table(name, manager_id, id) AS VALUES ('Mary', NULL, 1), ('John', 1, 2)", 2);
 
         assertFileSystemAccesses(
-                "SELECT child.age, parent.age FROM test_self_join_table child JOIN test_self_join_table parent ON child.parent = parent.id",
+                "SELECT employee.name, manager.name FROM test_self_join_table employee JOIN test_self_join_table manager ON employee.manager_id = manager.id",
```

Comment:
Why would it prevent DF from short-circuiting?

also, can we simply disable DF for the test query?

Reply:
@raunaqmorarka pointed out that this might be related to the extended stats causing the pruning.

Reply:
> might be related to the extended stats causing the pruning.

extended stats cannot cause pruning, since they are just estimates, rigt?

Reply:
> Do you rather prefer it this way?

yes

Reply:
> extended stats cannot cause pruning, since they are just estimates, right?

Correct. My bad. The split predicate is built from the `add` statistics from the transaction log.

https://github.com/trinodb/trino/blob/29ee2a541623071265b7c740caa3e796d946bf99/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitSource.java#L117-L128

https://github.com/trinodb/trino/blob/29ee2a541623071265b7c740caa3e796d946bf99/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L219-L222